src/inc/til/rectangle.h

```diff
@@ -0,0 +1,484 @@
+{
+    class rectangle
+    {
+    public:
+        constexpr rectangle() noexcept :
+            rectangle(til::point{ 0, 0 }, til::point{ 0, 0 })
+        {
+        }
+
+        // On 64-bit processors, int and ptrdiff_t are different fundamental types.
+        // On 32-bit processors, they're the same which makes this a double-definition
+        // with the `ptrdiff_t` one below.
+#if defined(_M_AMD64) || defined(_M_ARM64)
+        constexpr rectangle(int left, int top, int right, int bottom) noexcept :
+            rectangle(til::point{ left, top }, til::point{ right, bottom })
+        {
+        }
+#endif
+
+        rectangle(size_t left, size_t top, size_t right, size_t bottom) :
+            rectangle(til::point{ left, top }, til::point{ right, bottom })
+        {
+        }
+
+        constexpr rectangle(ptrdiff_t left, ptrdiff_t top, ptrdiff_t right, ptrdiff_t bottom) noexcept :
+            rectangle(til::point{ left, top }, til::point{ right, bottom })
+        {
+        }
+
+        // Creates a 1x1 rectangle with the given top-left corner.
+        rectangle(til::point topLeft) :
+            _topLeft(topLeft)
+        {
+            _bottomRight = _topLeft + til::point{ 1, 1 };
+        }
+
+        // Creates a rectangle where you specify the top-left corner (included)
+        // and the bottom-right corner (excluded)
+        constexpr rectangle(til::point topLeft, til::point bottomRight) noexcept :
+            _topLeft(topLeft),
+            _bottomRight(bottomRight)
+        {
+        }
+
+        // Creates a rectangle with the given size where the top-left corner
+        // is set to 0,0.
+        constexpr rectangle(til::size size) noexcept :
+            _topLeft(til::point{ 0, 0 }),
+            _bottomRight(til::point{ size.width(), size.height() })
+        {
+        }
+
+        // Creates a rectangle at the given top-left corner point X,Y that extends
+        // down (+Y direction) and right (+X direction) for the given size.
+        rectangle(til::point topLeft, til::size size) :
+            _topLeft(topLeft),
+            _bottomRight(topLeft + til::point{ size.width(), size.height() })
+        {
+        }
+
+#ifdef _WINCONTYPES_
+        // This extra specialization exists for SMALL_RECT because it's the only rectangle in the world that we know of
+        // with the bottom and right fields INCLUSIVE to the rectangle itself.
+        // It will perform math on the way in to ensure that it is represented as EXCLUSIVE.
+        rectangle(SMALL_RECT sr)
+        {
+            _topLeft = til::point{ static_cast<ptrdiff_t>(sr.Left), static_cast<ptrdiff_t>(sr.Top) };
+
+            _bottomRight = til::point{ static_cast<ptrdiff_t>(sr.Right), static_cast<ptrdiff_t>(sr.Bottom) } + til::point{ 1, 1 };
+        }
+#endif
+
+        // This template will convert to size from anything that has a Left, Top, Right, and Bottom field that appear convertable to an integer value

```

Comment:
```suggestion
        // This template will convert to size from anything that has a Left, Top, Right, and Bottom field that appear convertible to an integer value
```

Reply:
actually yeah also they all say `size` instead of rectangle

Reply:
Fixed both. Also fixed `convertable` --> `convertible` in `size.h` and `point.h` while I'm here.